modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ExecutableTableRegistryImpl.java

```diff
@@ -62,13 +62,19 @@ public ExecutableTableRegistryImpl(TableManager tableManager, SchemaManager sche
         this.clock = clock;
         this.tableCache = Caffeine.newBuilder()
                 .maximumSize(cacheSize)
-                .<Integer, ExecutableTable>buildAsync().asMap();
+                .<Long, ExecutableTable>buildAsync().asMap();
     }
 
     /** {@inheritDoc} */
     @Override
-    public CompletableFuture<ExecutableTable> getTable(int tableId, TableDescriptor tableDescriptor) {
-        return tableCache.computeIfAbsent(tableId, (k) -> loadTable(k, tableDescriptor));
+    public CompletableFuture<ExecutableTable> getTable(int schemaVersion, int tableId, TableDescriptor tableDescriptor) {
```

Comment:
do you think that TABLE_CACHE_SIZE==1024 is enough ? simultaneous work with such number of tables also without schema modification will raise all _loadTable_ further machinery, i have no clue what correct num there will be ok, still think about it ..

Reply:
As I understand, we introduce this cache to leverage impact of awaiting table availability, because of we have no schema synchronization yet.
I think, with schema synchronization local state will be always actual here. Thus, we could ask TableManager for a table by ID synchronously without any waiting, and cache will gone.